lib/events/auditlog.go

```diff
@@ -890,7 +901,37 @@ func (l *AuditLog) SearchSessionEvents(fromUTC, toUTC time.Time) ([]EventFields,
 		SessionEndEvent,
 	}
 
-	return l.SearchEvents(fromUTC, toUTC, query.Encode())
+	// because of the limit and distributed nature of auth server event
+	// logs, some events can be fetched with session end event and without
+	// session start event. to fix this, the code below filters out the events without
+	// start event to guarantee that all events in the range will get fetched
+	events, err := l.SearchEvents(fromUTC, toUTC, query.Encode(), limit)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	// filter out events that have start event, but don't have end event
```

Comment:
It took me a while to understand what is happening below after reading this comment. I'd word it like this to make it a bit clearer: `filter out 'session end' events that do not have a corresponding 'session start' event`.